exporter/signalfxexporter/dpclient.go

```diff
@@ -75,32 +74,28 @@ func (s *sfxDPClient) pushMetricsData(
 	ctx context.Context,
 	md pdata.Metrics,
 ) (droppedDataPoints int, err error) {
-	var numDroppedDataPoints int
-	var errs []error
-
 	rms := md.ResourceMetrics()
+	if rms.Len() == 0 {
+		return 0, nil
+	}
+
+	// All metrics in the pdata.Metrics will have the same access token because of the BatchPerResourceMetrics.
+	metricToken := s.retrieveAccessToken(rms.At(0))
+
+	var sfxDataPoints []*sfxpb.DataPoint
+
 	for i := 0; i < rms.Len(); i++ {
 		rm := rms.At(i)
 		if rm.IsNil() {
 			continue
 		}
```

Comment:
If we had debug asserts (if Go had them?) it would be useful to assert that `s.retrieveAccessToken(rm)==metricToken`. Maybe we need to add our own debug assert function (if it's even doable nicely in Go).
Nothing to do for now, but some food for thought.